presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -1449,7 +1451,7 @@ public static Signature resolveFunction(FunctionCall node, List<TypeSignaturePro
             Iterable<Expression> expressions,
             List<Expression> parameters)
     {
-        return analyzeExpressionsWithInputs(session, metadata, sqlParser, types, expressions, parameters).getExpressionTypes();
+        return analyzeExpressionsWithInputs(session, metadata, sqlParser, types, expressions, parameters, new NoOpWarningCollector()).getExpressionTypes();
```

Comment:
Why do these use NoOpWarningCollector?

Reply:
Not sure why this was done, possibly for tests where no warning collector is used. 
It seems that there should be an invocation of getExpressionTypes that takes WarningCollector as an parameter in case there are warnings added in the future.